### ISSUE TITLE
Max Session Expiry Interval in CONNECT does not work

### ISSUE THREAD

**Issue by ckrey** [emqx/emqx#1834]
#### Environment

- OS: Debian9
- Erlang/OTP: 19
- EMQ: 3.0 beta 2

#### Description

- case 2
CONNECT with a Session Expiry Interval of 42949672956 seconds (maximum)
DISCONNECT without setting a new Session Expiry Interval
Re-CONNECT with CleanStart = false
emqx does not indicate existing session, but should do so

```
19:34:28.165 [debug] MQTT(10.0.2.2:58925): RECV <<16,63,0,4,77,81,84,84,5,2,0,60,5,17,255,255,255,252,0,45,45,91,77,81,84,84,118,53,83,101,115,115,105,111,110,69,120,112,105,114,121,32,116,101,115,116,95,115,69,73,95,109,97,120,95,114,101,99,111,110,110,101,99,116,93>>
19:34:28.165 [debug] MQTT(@10.0.2.2:58925): RECV CONNECT(Q0, R0, D0, ClientId=-[MQTTv5SessionExpiry test_sEI_max_reconnect], ProtoName=MQTT, ProtoVsn=5, CleanStart=true, KeepAlive=60, Username=undefined, Password=undefined)
19:34:28.171 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58925): SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
19:34:28.171 [debug] MQTT(10.0.2.2:58925): SEND <<32,21,0,0,18,39,0,16,0,0,36,2,37,1,42,1,41,1,34,0,0,40,1>>
19:34:28.171 [debug] MQTT(10.0.2.2:58925): Keepalive at the interval of 45
19:34:29.165 [debug] MQTT(10.0.2.2:58925): RECV <<224,0>>
19:34:29.165 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58925): RECV DISCONNECT(Q0, R0, D0, ReasonCode=0)
19:34:29.165 [debug] MQTT(10.0.2.2:58925): Terminated for normal
19:34:29.165 [info] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58925): Shutdown for normal
19:34:29.663 [debug] MQTT(10.0.2.2:58927): RECV <<16,58,0,4,77,81,84,84,5,0,0,60,0,0,45,45,91,77,81,84,84,118,53,83,101,115,115,105,111,110,69,120,112,105,114,121,32,116,101,115,116,95,115,69,73,95,109,97,120,95,114,101,99,111,110,110,101,99,116,93>>
19:34:29.663 [debug] MQTT(@10.0.2.2:58927): RECV CONNECT(Q0, R0, D0, ClientId=-[MQTTv5SessionExpiry test_sEI_max_reconnect], ProtoName=MQTT, ProtoVsn=5, CleanStart=false, KeepAlive=60, Username=undefined, Password=undefined)
19:34:29.665 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58927): SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
19:34:29.665 [debug] MQTT(10.0.2.2:58927): SEND <<32,21,0,0,18,39,0,16,0,0,36,2,37,1,42,1,41,1,34,0,0,40,1>>
19:34:29.665 [debug] MQTT(10.0.2.2:58927): Keepalive at the interval of 45
19:34:30.670 [debug] MQTT(10.0.2.2:58927): RECV <<224,0>>
19:34:30.671 [debug] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58927): RECV DISCONNECT(Q0, R0, D0, ReasonCode=0)
19:34:30.671 [debug] MQTT(10.0.2.2:58927): Terminated for normal
19:34:30.672 [info] MQTT(-[MQTTv5SessionExpiry test_sEI_max_reconnect]@10.0.2.2:58927): Shutdown for normal
```

**Comment by Gilbert-Wong** [comment]
There are still some features in mqtt 5.0 not implemented. we plan to finish these work in beta3

**Comment by tigercl** [comment]
hi, @ckrey, it has been fixed in #1845  (comment)
